atlassian_jwt_auth/signer.py

```diff
@@ -54,10 +55,59 @@ def generate_jwt(self, audience, **kwargs):
             headers={'kid': key_identifier.key_id})
 
 
+class TokenReusingJWTAuthSigner(JWTAuthSigner):
+
+    def __init__(self, issuer, private_key_retriever, **kwargs):
+        super(TokenReusingJWTAuthSigner, self).__init__(
+            issuer, private_key_retriever, **kwargs)
+        self.reuse_threshold = kwargs.get('reuse_jwt_threshold', 0.95)
+
+    def get_cached_token(self):
+        return getattr(self, '_previous_token', None)
+
+    def set_cached_token(self, value):
+        self._previous_token = value
+
+    def can_reuse_token(self, existing_token, claims):
+        if existing_token is None:
+            return False
+        existing_claims = jwt.decode(existing_token, verify=False)
```

Comment:
Technically we don't need to call `jwt.decode` and we can modify the code to set the claims in addition to the token.